src/libraries/Common/src/System/TimeProvider.cs

```diff
@@ -10,115 +10,100 @@ namespace System
         /// </summary>
-        public DateTimeOffset LocalNow
+        public DateTimeOffset GetLocalNow()
         {
-            get
-            {
-                DateTime utcDateTime = UtcNow.UtcDateTime;
-                TimeSpan offset = LocalTimeZone.GetUtcOffset(utcDateTime);
-
-                long localTicks = utcDateTime.Ticks + offset.Ticks;
-                if ((ulong)localTicks > (ulong)s_maxDateTicks)
-                {
-                    localTicks = localTicks < s_minDateTicks ? s_minDateTicks : s_maxDateTicks;
-                }
+            DateTimeOffset utcDateTime = GetUtcNow();
+            TimeSpan offset = (LocalTimeZone ?? TimeZoneInfo.Local).GetUtcOffset(utcDateTime);
 
-                return new DateTimeOffset(localTicks, offset);
+            long localTicks = utcDateTime.Ticks + offset.Ticks;
+            if ((ulong)localTicks > (ulong)s_maxDateTicks)
+            {
+                localTicks = localTicks < s_minDateTicks ? s_minDateTicks : s_maxDateTicks;
             }
+
+            return new DateTimeOffset(localTicks, offset);
         }
 
         /// <summary>
         /// Gets a <see cref="TimeZoneInfo"/> object that represents the local time zone according to this <see cref="TimeProvider"/>'s notion of time.
         /// </summary>
-        public abstract TimeZoneInfo LocalTimeZone { get; }
+        /// <remarks>
+        /// The default implementation returns <see cref="TimeZoneInfo.Local"/>.
+        /// </remarks>
+        public virtual TimeZoneInfo LocalTimeZone { get => TimeZoneInfo.Local; }
 
         /// <summary>
         /// Gets the frequency of <see cref="GetTimestamp"/> of high-frequency value per second.
         /// </summary>
-        public long TimestampFrequency { get; }
-
-        /// <summary>
-        /// Creates a <see cref="TimeProvider"/> that provides a clock based on <see cref="DateTimeOffset.UtcNow"/>,
-        /// a time zone based on <paramref name="timeZone"/>, a high-performance time stamp based on <see cref="Stopwatch"/>,
-        /// and a timer based on <see cref="Timer"/>.
-        /// </summary>
-        /// <param name="timeZone">The time zone to use in getting the local time using <see cref="LocalNow"/>. </param>
-        /// <returns>A new instance of <see cref="TimeProvider"/>. </returns>
-        /// <exception cref="ArgumentNullException"><paramref name="timeZone"/> is null.</exception>
-        public static TimeProvider FromLocalTimeZone(TimeZoneInfo timeZone)
-        {
-#if SYSTEM_PRIVATE_CORELIB
-            ArgumentNullException.ThrowIfNull(timeZone);
-#else
-            if (timeZone is null)
-            {
-                throw new ArgumentNullException(nameof(timeZone));
-            }
-#endif // SYSTEM_PRIVATE_CORELIB
-
-            return new SystemTimeProvider(timeZone);
-        }
+        /// <remarks>
+        /// The default implementation returns <see cref="Stopwatch.Frequency"/>.
```

Comment:
Maybe add a comment that the value should be idempotent and never change for a given TimeProvider instance.